packages/runtime/src/components/builtin/Box/animations.tsx

```diff
@@ -1,365 +1,133 @@
-  const boxAnimateDuration = useMediaQuery(props.boxAnimateDuration) || DEFAULT_BOX_ANIMATE_DURATION
-  const boxAnimateDelay = useMediaQuery(props.boxAnimateDelay) || DEFAULT_BOX_ANIMATE_DELAY
-  const itemAnimateType = useMediaQuery(props.itemAnimateType) || DEFAULT_ITEM_ANIMATE_TYPE
-  const itemAnimateDuration =
-    useMediaQuery(props.itemAnimateDuration) || DEFAULT_ITEM_ANIMATE_DURATION
-  const itemAnimateDelay = useMediaQuery(props.itemAnimateDelay) || DEFAULT_ITEM_ANIMATE_DELAY
-  const itemStaggerDuration =
-    useMediaQuery(props.itemStaggerDuration) || DEFAULT_ITEM_STAGGER_DURATION
-
-  const isBoxVisible = useElementOnScreen(boxElement, {
-    root: null,
-    rootMargin: `0px 0px -10% 0px`,
-    threshold: 0.2,
-  })
-
-  const itemControls = useAnimation()
-  const boxControls = useAnimation()
+const enteredBoxAnimationProperties: { [key in BoxAnimateIn]: CSSObject } = {
+  none: { opacity: 1 },
+  fadeIn: { opacity: 1 },
+  fadeLeft: { transform: 'translate3d(0px,0,0)', opacity: 1 },
+  fadeRight: { transform: 'translate3d(0px,0,0)', opacity: 1 },
+  fadeDown: { transform: 'translate3d(0,0px,0)', opacity: 1 },
+  fadeUp: { transform: 'translate3d(0,0px,0)', opacity: 1 },
+  blurIn: { filter: 'blur(0px)', opacity: 1 },
+  scaleDown: {
+    transform: 'scale(1)',
+    opacity: 1,
+  },
+  scaleUp: {
+    transform: 'scale(1)',
+    opacity: 1,
+  },
+}
 
-  const setSequence = useCallback(
-    (variant: keyof BoxAnimationVariants) => {
-      boxControls.stop()
-      itemControls.stop()
-      boxControls.set(variant)
-      itemControls.set(variant)
+export function useBoxAnimation(
+  responsiveAnimationType: ResponsiveValue<BoxAnimateIn> | undefined,
+  responsiveDuration: ResponsiveValue<number> | undefined,
+  responisveDelay: ResponsiveValue<number> | undefined,
+  itemResponsiveAnimationType: ResponsiveValue<BoxAnimateIn> | undefined,
+) {
+  const reducedMotion = usePrefersReducedMotion()
+  const animationType = useMediaQuery(responsiveAnimationType) || DEFAULT_BOX_ANIMATE_TYPE
+  const itemAnimationType = useMediaQuery(itemResponsiveAnimationType) || DEFAULT_ITEM_ANIMATE_TYPE
+  const duration = useMediaQuery(responsiveDuration) || DEFAULT_BOX_ANIMATE_DURATION
+  const delay = useMediaQuery(responisveDelay) || DEFAULT_BOX_ANIMATE_DELAY
+  const actualDelay = delay * 1000
+  const actualDuration = duration * 1000
+
+  const className = useStyle({
+    ...exitedBoxAnimationProperties[animationType],
+    transition: `all 0ms`,
+    '&.show': {
+      ...enteredBoxAnimationProperties[animationType],
+      transition: `transform ${actualDuration}ms cubic-bezier(0.16, 0.84, 0.44, 1) ${actualDelay}ms,filter ${actualDuration}ms cubic-bezier(0.16, 0.84, 0.44, 1) ${actualDelay}ms, opacity ${actualDuration}ms ease ${actualDelay}ms`,
+    },
+    '&:not(.show) > div > div': {
+      ...exitedBoxAnimationProperties[itemAnimationType],
+      transition: `all 0ms !important`,
```

Comment:
this has to be important because the other css class is more specific
This css has to be specified here since it is based on a class on the root.
The item css has to be specified there since it needs the index to know how much stagger to add.